### DISCUSSION TITLE
Prisma integration

### DISCUSSION THREAD

**Discussion by RobertCraigie** [fastapi-users/fastapi-users#821] [Ideas]
Hi, I would love to add support for Prisma Client Python to this project!

Source code: https://github.com/RobertCraigie/prisma-client-py
Documentation: https://prisma-client-py.readthedocs.io/en/stable/

There is one limitation that must be discussed first:

Models are not defined in a python file, they are instead defined in a Prisma schema file, this means that all the fields used by `fastapi-users` must be defined by the user. We could still include the base model in the documentation that could then be copy-pasted.

For example:

```prisma
model User {
  id           String  @default(uuid())
  email        String  @unique
  is_active    Boolean @default(true)
  is_superuser Boolean @default(false)
  is_verified  Boolean @default(false)

  @@index(email)
}
```

Prisma Schema reference: https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference

If the aforementioned limitation is not a dealbreaker, I would be happy to create the integration myself as the maintainer of the ORM.

I'll also happily answer any questions if needed!

**Comment by frankie567** [comment]
Hi @RobertCraigie!

Why not! Actually, if you look at [some DB adapters](https://github.com/fastapi-users/fastapi-users-db-tortoise/blob/main/fastapi_users_db_tortoise/__init__.py), you'll see that they also have some field definitions duplication to make them compatible.

What I suggest you is to use the Cookiecutter for DB adapters: https://github.com/fastapi-users/fastapi-users-db-cookiecutter It'll generate you all the code to help you get started.

Once you release it, I'll be happy to add it to the doc as a community-contributed adapter.

Cheers!
